ts/components/services/NewServiceListItem.tsx

```diff
@@ -0,0 +1,62 @@
+import I18n from "i18n-js";
+import * as pot from "italia-ts-commons/lib/pot";
+import * as React from "react";
+import { StyleSheet } from "react-native";
+import { ServicePublic } from "../../../definitions/backend/ServicePublic";
+import { getEnabledChannelsForService } from "../../screens/preferences/common";
+import { ProfileState } from "../../store/reducers/profile";
+import customVariables from "../../theme/variables";
+import ListItemComponent from "../screens/ListItemComponent";
+
+type Props = Readonly<{
+  item: pot.Pot<ServicePublic, Error>;
+  profile: ProfileState;
+  onSelect: (service: ServicePublic) => void;
+  isRead: boolean;
+  hideSeparator: boolean;
+}>;
+
+const styles = StyleSheet.create({
+  listItem: {
+    marginLeft: customVariables.contentPadding,
+    marginRight: customVariables.contentPadding
+  }
+});
+
+export default class NewServiceListItem extends React.PureComponent<Props> {
+  public render() {
+    const potService = this.props.item;
+    const onPress = pot.getOrElse(
```

Comment:
render function is called at every change of state
Can we avoid to create a function each time?
I'm thinking about a solution like this, what do you think about it?

```javascript
private handleItemPressed(potService: pot.Pot<ServicePublic, Error>): void {
    if (pot.isSome(potService)) {
      this.props.onSelect(potService.value);
    }
  }
```

Reply:
Examining the component, it seems we cannot avoid calling an arrow function when declaring onPress: the function we have to pass to the component wants the selected service as input while the onPress attribute wants a function with no inputs. For the case you mentioned, it need to be:
 `onPress={() => this.handleItemPressed(potService)}` 

I will look to search if any other solution exist to overcome it.

Reply:
Yes you are right!
arrow function returns a new function every time
So if don't find any better solution, leave the code as it is